features/scheduled_task_executor.feature

```diff
@@ -0,0 +1,36 @@
+Feature: Verifies that unhandled exceptions in scheduled tasks are reported to Bugsnag regardless of Scheduler configuration
+
+Scenario: By default a single-threaded ThreadPoolTaskScheduler is used
+    When I run spring boot "ScheduledTaskScenario" with the defaults
+    Then I should receive a request
+    And the request is a valid for the error reporting API
+    And the exception "message" equals "Unhandled exception from ScheduledTaskService"
+
+Scenario: A TaskScheduler supplied as a bean is used
+    Given I set environment variable "custom_task_scheduler_bean" to "true"
+    And I run spring boot "ScheduledTaskScenario" with the defaults
+    Then I should receive a request
+    And the request is a valid for the error reporting API
+    And the exception "message" equals "Unhandled exception from ScheduledTaskService"
+
+Scenario: A ScheduledExecutorService supplied as a bean is used
+    Given I set environment variable "scheduled_executor_service_bean" to "true"
+    And I run spring boot "ScheduledTaskScenario" with the defaults
+    Then I should receive a request
+    And the request is a valid for the error reporting API
+    And the exception "message" equals "Unhandled exception from ScheduledTaskService"
+
+Scenario: For multiple ScheduledExecutorService beans, the one named "taskScheduler" is used
```

Comment:
Can we run the same test for a multiple `TaskScheduler` beans as well? I think that will complete the cover all scenarios.